### DISCUSSION TITLE
How to upload a Webdriver IO project (JavaScript) to a cloud service

### DISCUSSION THREAD

**Discussion by sergeykatelyna** [webdriverio/webdriverio#8357] [Q&A] [answered]
Hello

I am trying to find a way to run Webdriver IO automation tests (JavaScript) on a cloud service, so that I can upload the code there and trigger it by 1 API request.

I see alternative 2 approaches to this task:
1. A cloud service where I can upload the Webdriver IO project (JavaScript) and trigger the tests by 1 API request. **Please advise if you know such cloud services?**
2. Somehow export the tests from the Webdriver IO project (JavaScript) as a ".side" file (it is Selenium IDE export format). And after that, upload this ".side" file to a service like Testingbot. **Please advise if it is even possible to export tests from a Webdriver IO project as a ".side" file (or in any other format that can be uploaded to a could service)?**

Thanks in advance.

**Comment by christian-bromann** [comment, marked as answer]
> 1. Please advise if you know such cloud services?

I don't. Cloud services usually work differently.

> 2\. Please advise if it is even possible to export tests from a Webdriver IO project as a ".side" file

The Selenium IDE doesn't support WebdriverIO as output format.

**Comment by sergeykatelyna** [comment]
@christian-bromann Hey, thanks for the answer. I wanted to clarify this.

When I run WDIO tests through Testingbot, the tests are sent there as a list of simple commands.
![image](https://user-images.githubusercontent.com/34918999/169290014-37efce24-35b4-4d9a-87f4-c405ffe56d23.png)

So, I wonder if it is possible to pull these simple commands (from the WDIO project, from Testingbot, or using some other app) and then upload them to a service like Testingbot.

**Reply by christian-bromann** [reply]
Pull the commands from Testingbot and then upload them to Testingbot?

**Comment by sergeykatelyna** [comment]
@christian-bromann Yeah)

There is no option in Testingbot to run a finished WDIO test for the second time directly from their service. And, as far as I know, there is no option to pull simple commands of a finished test (other than manual copying).

So, all I want is to get simple commands of a WDIO test from somewhere and upload them to Testingbot or any other service. So that I can trigger it by 1 API request, without the need to deploy a WDIO project on my machine and trigger the test from there.

Here is how you can upload a test in ".side" format (Selenium IDE) to Testingbot:
![image](https://user-images.githubusercontent.com/34918999/169296383-0249478a-6d0d-4a9e-94e4-5b72d8638861.png)

**Reply by christian-bromann** [reply]
> There is no option in Testingbot to run a finished WDIO test for the second time directly from their service.

You might want to reach out to the Testingbot team and ask for support of WebdriverIO.

> Here is how you can upload a test in ".side" format (Selenium IDE) to Testingbot:

Again, WebdriverIO is not supported by the Selenium IDE

**Reply by sergeykatelyna** [reply]
Thanks
